android/titanium/src/java/org/appcelerator/titanium/TiApplicationLifecycle.java

```diff
@@ -29,19 +30,49 @@ public void onActivityCreated(Activity activity, Bundle savedInstanceState)
 	@Override
 	public void onActivityStarted(Activity activity)
 	{
-		if (activityCount == 0 && tiApp != null && tiApp.isAnalyticsEnabled()) {
-			APSAnalytics.getInstance().sendAppForegroundEvent();
+		// If no activities have been started, then app is going to be put into the foreground.
+		if (this.activityCount == 0) {
+			// Fire Ti.App resume events.
+			KrollModule appModule = this.tiApp.getModuleByName("App");
```

Comment:
Should we also put `appModule` in `TiApplicationLifecycle`, so we don't have to grab it each time?

Reply:
I noticed that we add a new "App" module instance every time a new JavaScript runtime gets created (ie: back-out of the app and relaunch).